src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Usage/DetectPLINQNopsTest.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+using VerifyCS = Test.Utilities.CSharpCodeFixVerifier<
+    Microsoft.NetCore.Analyzers.Usage.DetectPLINQNopsAnalyzer,
+    Microsoft.NetCore.CSharp.Analyzers.Usage.DetectPLINQNopsFixer>;
+
+namespace Microsoft.NetCore.Analyzers.Usage.UnitTests
+{
+    public sealed class DetectPLINQNopsTest
+    {
+        [Fact]
+        public async Task AsParallelToListInForeach_SingleDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Test() { foreach(var s in {|#0:Enumerable.Range(0,1).Select(x => x*2).AsParallel().ToList()|});}
+    }
+}", VerifyCS.Diagnostic(DetectPLINQNopsAnalyzer.DefaultRule).WithLocation(0).WithArguments("Enumerable.Range(0,1).Select(x => x*2).AsParallel().ToList()"));
+        }
+
+        [Fact]
+        public async Task AsParallelAtStart_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+        public void Test() { foreach(var s in Enumerable.Range(0,1).AsParallel().Select(x => x*2).ToList());}
+    }
+}");
+        }
+
+        [Fact]
+        public async Task AsParallelIsRoot_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.Collections.Generic;
+using System.Linq;
+namespace ConsoleApplication1
+{
+    class TypeName
+    {   
+            IEnumerable<int> AsParallel() => Enumerable.Empty<int>();
```

Comment:
There is no local function with vbnet so depending on what exactly you want to test, you can either create a `Func` or a `Private Function`.

Reply:
This test was initially to increase coverage for the paths where i did not yet know whether this is the correct symbol. I will remove this test, it does nothing now